File: admin-backend/admin-system/src/main/java/io/github/faustofan/admin/system/infrastructure/reponsitory/SysRoleRepository.java
package io.github.faustofan.admin.system.infrastructure.reponsitory;

import io.github.faustofan.admin.system.domain.model.SysRole;
import io.github.faustofan.admin.system.domain.model.SysRoleTable;
import io.github.faustofan.admin.system.domain.model.Tables;
import org.babyfish.jimmer.spring.repository.JRepository;
import org.springframework.stereotype.Repository;

/**
 * 系统角色仓库
 * 提供对 SysRole 实体的数据库操作方法
 */
@Repository
public interface SysRoleRepository extends JRepository<SysRole, Long> {

    SysRoleTable table = Tables.SYS_ROLE_TABLE;

}

File: src/main/kotlin/io/github/faustofan/admin/system/infrastructure/persistence/SysRoleRepository.kt
package io.github.faustofan.admin.system.infrastructure.persistence

import io.github.faustofan.admin.system.domain.model.SysRole
import org.babyfish.jimmer.spring.repository.KRepository
import org.springframework.stereotype.Repository

@Repository
interface SysRoleRepository : KRepository<SysRole, Long> {

    // 根据角色编码查询
//    fun findByCode(code: String): SysRole?

    // 查询某个用户的所有角色 (通过反向关联)
    // Jimmer 会自动生成 JOIN SQL
//    fun findByUsersUsername(username: String): List<SysRole>
}